tests/jobs/test_scheduler_job.py

```diff
@@ -3299,3 +3299,60 @@ def test_task_with_upstream_skip_process_task_instances(self):
             assert tis[dummy2.task_id].state == State.SUCCESS
             # dummy3 should be skipped because dummy1 is skipped.
             assert tis[dummy3.task_id].state == State.SKIPPED
+
+
+class TestSchedulerJobQueriesCount(unittest.TestCase):
+    """
+    These tests are designed to detect changes in the number of queries for
+    different DAG files. These tests allow easy detection when a change is
+    made that affects the performance of the SchedulerJob.
+    """
+    def setUp(self) -> None:
+        clear_db_runs()
+        clear_db_pools()
+        clear_db_dags()
+        clear_db_sla_miss()
+        clear_db_errors()
+
+    @parameterized.expand(
+        [
+            # pylint: disable=bad-whitespace
+            # expected, dag_count, task_count, start_ago, schedule_interval, shape
+            # One DAG with one task per DAG file
+            ( 13,  1,  1, "1d",   "30m",  "no_structure"),  # noqa
+            # One DAG with five tasks per DAG  file
+            ( 25,  1,  5, "1d",   "30m",  "no_structure"),  # noqa
+            # 10 DAGs with 10 tasks per DAG file
+            (108, 10, 10, "1d",   "30m",  "no_structure"),  # noqa
```

Comment:
All tasks are set to SCHEDULED so no reason to test different shapes. The result depends only on the number of tasks in DAG